lib/puppet/provider/logical_volume/lvm.rb

```diff
@@ -110,7 +115,14 @@ def create
             args.push('--type', @resource[:type])
         end
 
-        args << @resource[:volume_group]
+	if @resource[:thin]
```

Comment:
if `@resource[:thin]` is `:false` or `'false'` then this will be truthy. The type should either munge or only accept literal `true` and `false` values.